lib/tasks/prune_reports.rake

```diff
@@ -100,13 +100,25 @@ UNITS:
         puts "#{start_time.to_s(:db)}: Deleting #{deletion_count} orphaned records from #{model.table_name}"
         pbar = ProgressBar.new('Deleting', deletion_count, STDOUT)
 
-        # Deleting a very large group of records in MySQL can be very slow with no feedback
+        struct = database_settings
+        adapter = struct.adapter
+
+        # Deleting a very large group of records can be very slow with no feedback
         # Breaking the deletion up into blocks turns out to be overall faster
         # and allows for progress feedback
         DELETION_BATCH_SIZE = 1000
         while deletion_count > 0
+          case adapter
+            when 'mysql', 'mysql2', 'sqlite3'
+              command_line = "delete from #{model.table_name} where #{deletion_where_clause} limit #{DELETION_BATCH_SIZE}"
+            when 'postgresql'
+              command_line = "delete from #{model.table_name} where #{deletion_where_clause} DESC LIMIT #{DELETION_BATCH_SIZE}"
```

Comment:
I don't get it - the DESC keyboard should only be allowed after an ORDER BY, and there isn't any ORDER BY in this query. http://www.postgresql.org/docs/9.1/static/sql-select.html

I'm curious what version of PostgreSQL are you running?